llvm/test/Transforms/LoopVectorize/conditional-scalar-assignment.ll

```diff
@@ -0,0 +1,2415 @@
+; NO-EVL-NEXT:    [[TMP27:%.*]] = icmp sge i32 [[TMP26]], 0
+; NO-EVL-NEXT:    [[TMP28:%.*]] = select i1 [[TMP27]], float [[CSA_EXTRACT]], float 1.000000e+00
+; NO-EVL-NEXT:    [[CMP_N:%.*]] = icmp eq i64 [[WIDE_TRIP_COUNT]], [[N_VEC]]
+; NO-EVL-NEXT:    br i1 [[CMP_N]], label [[EXIT_LOOPEXIT:%.*]], label [[SCALAR_PH]]
+; NO-EVL:       scalar.ph:
+; NO-EVL-NEXT:    [[BC_RESUME_VAL:%.*]] = phi i64 [ [[N_VEC]], [[MIDDLE_BLOCK]] ], [ 0, [[FOR_BODY_PREHEADER]] ]
+; NO-EVL-NEXT:    br label [[FOR_BODY:%.*]]
+; NO-EVL:       exit.loopexit:
+; NO-EVL-NEXT:    [[T_1_LCSSA:%.*]] = phi float [ [[T_1:%.*]], [[FOR_BODY]] ], [ [[TMP28]], [[MIDDLE_BLOCK]] ]
+; NO-EVL-NEXT:    [[S_1_LCSSA:%.*]] = phi float [ [[S_1:%.*]], [[FOR_BODY]] ], [ [[TMP20]], [[MIDDLE_BLOCK]] ]
+; NO-EVL-NEXT:    [[TMP29:%.*]] = fadd float [[T_1_LCSSA]], [[S_1_LCSSA]]
+; NO-EVL-NEXT:    br label [[EXIT]]
+; NO-EVL:       exit:
+; NO-EVL-NEXT:    [[ADD:%.*]] = phi float [ [[TMP29]], [[EXIT_LOOPEXIT]] ], [ 2.000000e+00, [[ENTRY:%.*]] ]
+; NO-EVL-NEXT:    ret float [[ADD]]
+; NO-EVL:       for.body:
+; NO-EVL-NEXT:    [[IV:%.*]] = phi i64 [ [[BC_RESUME_VAL]], [[SCALAR_PH]] ], [ [[IV_NEXT:%.*]], [[FOR_BODY]] ]
+; NO-EVL-NEXT:    [[S_021:%.*]] = phi float [ 1.000000e+00, [[SCALAR_PH]] ], [ [[S_1]], [[FOR_BODY]] ]
+; NO-EVL-NEXT:    [[T_020:%.*]] = phi float [ 1.000000e+00, [[SCALAR_PH]] ], [ [[T_1]], [[FOR_BODY]] ]
+; NO-EVL-NEXT:    [[ARRAYIDX:%.*]] = getelementptr inbounds float, ptr [[DATA0]], i64 [[IV]]
+; NO-EVL-NEXT:    [[TMP30:%.*]] = load float, ptr [[ARRAYIDX]], align 4
+; NO-EVL-NEXT:    [[CMP1:%.*]] = fcmp ogt float [[TMP30]], 0.000000e+00
+; NO-EVL-NEXT:    [[T_1]] = select i1 [[CMP1]], float [[TMP30]], float [[T_020]]
+; NO-EVL-NEXT:    [[ARRAYIDX5:%.*]] = getelementptr inbounds float, ptr [[DATA1]], i64 [[IV]]
+; NO-EVL-NEXT:    [[TMP31:%.*]] = load float, ptr [[ARRAYIDX5]], align 4
+; NO-EVL-NEXT:    [[CMP6:%.*]] = fcmp ogt float [[TMP31]], 0.000000e+00
+; NO-EVL-NEXT:    [[S_1]] = select i1 [[CMP6]], float [[TMP31]], float [[S_021]]
+; NO-EVL-NEXT:    [[IV_NEXT]] = add nuw nsw i64 [[IV]], 1
+; NO-EVL-NEXT:    [[EXITCOND_NOT:%.*]] = icmp eq i64 [[IV_NEXT]], [[WIDE_TRIP_COUNT]]
+; NO-EVL-NEXT:    br i1 [[EXITCOND_NOT]], label [[EXIT_LOOPEXIT]], label [[FOR_BODY]], !llvm.loop [[LOOP13:![0-9]+]]
+;
+entry:
+  %cmp19 = icmp sgt i32 %N, 0
+  br i1 %cmp19, label %for.body.preheader, label %exit
+
+for.body.preheader:                               ; preds = %entry
+  %wide.trip.count = zext i32 %N to i64
+  br label %for.body
+
+exit.loopexit:                        ; preds = %for.body
+  %0 = fadd float %t.1, %s.1
+  br label %exit
+
+exit:                                 ; preds = %exit.loopexit, %entry
+  %add = phi float [ %0, %exit.loopexit ], [ 2.000000e+00, %entry ]
+  ret float %add
+
+for.body:                                         ; preds = %for.body.preheader, %for.body
```

Comment:
nit: s/for.body/loop/, s/for.body.preheader/loop.ph/. Can `exit.loopexit` be merged into `exit`?

Reply:
> Can exit.loopexit be merged into exit

I don't think it makes sense. Moving the `fadd` into `exit` is weird. What should the phi values be coming from entry? And we don't want to add an unconditional add into `exit` if we don't enter the loop.